lib/register.js

```diff
@@ -0,0 +1,235 @@
+    if (!(category in this.categories)) {
+      this.categories[category] = [];
+    }
+    this.categories[category].push(`${manKey}/${fixKey}`);
+  }
+
+  /**
+   * @private
+   * @param {string} manKey The manufacturer key.
+   * @param {string} fixKey The fixture (redirect) key.
+   * @param {string} contributor The contributor to add this fixture to.
+   */
+  _addFixtureToContributor(manKey, fixKey, contributor) {
+    if (!(contributor in this.contributors)) {
+      this.contributors[contributor] = [];
+    }
+    this.contributors[contributor].push(`${manKey}/${fixKey}`);
+  }
+
+  /**
+   * @returns {object} The sorted register.
+   */
+  getAsSortedObject() {
+    const sortedRegister = {
+      filesystem: getObjectSortedByKeys(this.filesystem),
+      manufacturers: getObjectSortedByKeys(this.manufacturers, arr => arr.sort(localeSort)),
+      categories: getObjectSortedByKeys(this.categories, arr => arr.sort(localeSort)),
+      contributors: {},
+      rdm: {},
+      lastUpdated: []
+    };
+
+    // copy sorted contributors into register
+    const sortedContributors = Object.keys(this.contributors).sort((a, b) => {
+      const fixturesDelta = this.contributors[b].length - this.contributors[a].length;
+      const nameDelta = a > b ? 1 : a < b ? -1 : 0;
+      return fixturesDelta !== 0 ? fixturesDelta : nameDelta;
+    });
+    sortedContributors.forEach(contributor => {
+      sortedRegister.contributors[contributor] = this.contributors[contributor].sort(localeSort);
+    });
+
+    // the higher the value, the higher the rank if the dates are the same
+    const lastActionRankMapping = {
+      'modified': 10,
+      'imported': 20,
+      'created': 30
+    };
+
+    // add fixture list sorted by lastModifyDate
+    sortedRegister.lastUpdated = Object.keys(this.filesystem).filter(
+      fixKey => `lastModifyDate` in this.filesystem[fixKey]
+    ).sort((a, b) => {
+      const fixA = this.filesystem[a];
+      const fixB = this.filesystem[b];
+      const dateDelta = new Date(fixB.lastModifyDate) - new Date(fixA.lastModifyDate);
+      const actionDelta = lastActionRankMapping[fixB.lastAction] - lastActionRankMapping[fixA.lastAction];
+      const keyDelta = a > b ? 1 : a < b ? -1 : 0;
+      return dateDelta !== 0 ? dateDelta : (actionDelta !== 0 ? actionDelta : keyDelta);
+    });
+
+    // copy sorted RDM data into register
+    for (const manId of Object.keys(this.rdm).sort(localeSort)) {
+      sortedRegister.rdm[manId] = {
+        key: this.rdm[manId].key,
+        models: getObjectSortedByKeys(this.rdm[manId].models)
+      };
+    }
+
+    return sortedRegister;
+  }
+}
+
+/**
+ *
+ * @param {object} obj The object to sort.
+ * @param {function|null} itemSortFunction A function to be invoked with the value of an object entry to sort it (useful for array values).
+ * @returns {object} A new object with the same entries, sorted by keys.
+ */
+function getObjectSortedByKeys(obj, itemSortFunction) {
```

Comment:
If the parameter is called `itemSortFunction`, I'd expected an `(a, b) => ...` function here (and no mapping function). So either rename to `itemCallback` (or so) or directly accept the `(a, b) => ...` function.